docs/0002-atlas-translations-management.rst

```diff
@@ -0,0 +1,162 @@
+``edx-platform.po`` and ``edx-platform-js.po`` files.
+More details are available in the `edx-platform resources combination decision document`_.
+
+The `openedx-app-ios`_ uses a clean architecture. Therefore, each clean architecture module has its own i18n
+localizable strings file. This is useful for Developer Experience.
+However, it's not good for Translator Experience because translators
+often are unaware of the application engineering architecture and would like to be able to quickly search for strings
+regardless of their location in the code.
+
+Decision
+********
+
+This decision requires a few changes to the mobile app translation workflow:
+
+- Translation files are no longer comitted directly to the mobie app repositories (`openedx-app-ios`_ / `openedx-app-android`_)
+- Keep the i18n string files split into each module without hurting Developer Experience.
+- String files are combined into a single file before uploading it to Transifex
+  (through the `openedx-translations repository`_)
+- Translators can translate the entire app within a single transifex resource
+- Before releasing to App Store, developers need to run a new ``make pull_translations`` command
+- After translations are pulled the translation files are split into each module
+  as if it has been individual resource (a new script)
+
+
+String file combination and splitting process
+=============================================
+This is a new process that's being introduced to have the best combination
+of Developer Experience and Translator Experience.
+
+Translators need to have less resources as possible,
+while Engineers need to have clean architecture with strings split into each app to aid maintainability.
+
+Combining the files during ``combine_translations``
+---------------------------------------------------
+
+Combining the string files to aid Translator Experience in the following manner:
+
+Suppose we have two modules with one Localizable.strings file in each::
+
+  Course/Course/en.lproj/Localizable.strings
+    "COURSE_CONTAINER.VIDEOS" =  "Videos";
+    "COURSE_CONTAINER.DATES" =  "Dates";
+    "COURSE_CONTAINER.DISCUSSIONS" =  "Discussions";
+    "COURSE_CONTAINER.HANDOUTS" =  "Handouts";
+    "COURSE_CONTAINER.HANDOUTS_IN_DEVELOPING" =  "Handouts In developing";
+
+    "HANDOUTS_CELL_HANDOUTS.TITLE" = "Handouts";
+    "HANDOUTS_CELL_ANNOUNCEMENTS.TITLE" = "Announcements";
+
+  Dashboard/Dashboard/en.lproj/Localizable.strings:
+    "TITLE" = "Dashboard";
+    "HEADER.COURSES" = "Courses";
+    "HEADER.WELCOME_BACK" = "Welcome back. Let's keep learning.";
+
+    "EMPTY.TITLE" = "It's empty";
+    "EMPTY.SUBTITLE" = "You are not enrolled in any courses yet.";
+
+This combine translations script will collect the strings, remove unneeded comments and combine them in a temporary
+file while prefixing each entry with its module name. This will make splitting clear and
+avoid collision in string IDs::
```

Comment:
I know you meant code comments when saying "remove unneeded comments", but it reminded me of https://github.com/openedx/frontend-build/pull/460

Do we have a plan in place to support message descriptions for translators in transifex?

Reply:
I'm unware of how Translators comments work in iOS, but definilty it should be possible to support those comments.

There's no text processing limitation that would prevent us from pushing the comments as long as they're present in the `en.lproj/Localizable.strings`